compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1303,6 +1303,8 @@ OMR::ResolvedMethodSymbol::genIL(TR_FrontEnd * fe, TR::Compilation * comp, TR::S
       _methodFlags.set(IlGenSuccess, false);
       if (optimizer) //if the exception is from ilgen opts we need to restore previous optimizer
          comp->setOptimizer(previousOptimizer);
+      else  //if the exception is from genIL() we need to reset IlGenerator to 0 
```

Comment:
Rather than this comment which mirrors what the code is doing, I find the rationale in your commit message to be a far more useful comment here.  Please include that.